dev/io.openliberty.microprofile.health.4.0.internal/src/io/openliberty/microprofile/health40/internal/HealthFileUtils.java

```diff
@@ -0,0 +1,251 @@
+    private static volatile File readyFile;
+    private static volatile File liveFile;
+
+    public static File getHealthDirFile() {
+
+        if (healthDirFile == null) {
+            File serverConfigDirFile = new File(System.getProperty("server.config.dir"));
+            healthDirFile = new File(serverConfigDirFile, "health");
+        }
+
+        return healthDirFile;
+    }
+
+    public static File getStartFile() {
+
+        if (startedFile == null) {
+            File healthDirFile = getHealthDirFile();
+            startedFile = new File(healthDirFile, HealthCheckFileName.STARTED_FILE.getFileName());
+
+        }
+        return startedFile;
+    }
+
+    public static File getReadyFile() {
+
+        if (readyFile == null) {
+            File healthDirFile = getHealthDirFile();
+            readyFile = new File(healthDirFile, HealthCheckFileName.READY_FILE.getFileName());
+
+        }
+        return readyFile;
+    }
+
+    public static File getLiveFile() {
+
+        if (liveFile == null) {
+            File healthDirFile = getHealthDirFile();
+            liveFile = new File(healthDirFile, HealthCheckFileName.LIVE_FILE.getFileName());
+
+        }
+        return liveFile;
+    }
+
+    /*
+     * Determine if this is a valid system.
+     * Utility class will not keep state, up to the caller to remember.
+     */
+    /**
+     * This is called upon activation of HealthCheckService (i.e. HealthCheck40ServiceImpl and up).
+     * This determines if a system is valid for File Health check by checking write permissions and "sanitizing"
+     * the existing location of any previous files caused by a server crash.
+     *
+     *
+     * - Check if the /health directory exists (at server.config.dir)
+     * -NO
+     * -- Check if we can create the /health directory
+     * --- YES
+     * ---- Check if we can write to directory
+     * -----YES == all good
+     * -----NO == issue warning that we cannot write to /health directory
+     * --- NO Issue warning that we cannot create /health directory
+     * -YES
+     * --Check if existing 'started', 'live', 'ready' files exist.
+     * --YES
+     * ---Attempt to delete
+     * ----If the file is a Directory => issue warning about directory with matching name. Can not delete as this may contain user data (created by user).
+     * ---Issue warning if we can not delete.
+     *
+     * @return boolean if this is a valid system for file health check functionality
+     * @throws IOException
+     */
+    public static boolean isValidSystem() throws IOException {
+        healthDirFile = getHealthDirFile();
+
+        //Health Dir does not exist -> create and test write
+        if (!healthDirFile.exists()) {
+            if (!HealthFileUtils.createDirectory(healthDirFile)) {
+                //TODO: Create the real message ID
+                Tr.warning(tc, "Unable to create the /health directory. The mpHealth file-based health check reporting will be disabled.");
```

Comment:
Should all the warning msgs, be debug msgs for beta, and then create new messages with messageIDs later?

Reply:
fixed with https://github.com/OpenLiberty/open-liberty/pull/30811/commits/96f6b7f3a6322a9f5290bc471b75ec0c4028c4dc